packages/eslint-plugin/typings/eslint-rules.d.ts

```diff
@@ -195,7 +195,7 @@ declare module 'eslint/lib/rules/key-spacing' {
 
 declare module 'eslint/lib/rules/keyword-spacing' {
   import type { TSESLint, TSESTree } from '@typescript-eslint/utils';
-  import type { RuleFunction } from '@typescript-eslint/utils/dist/ts-eslint';
+  import type { RuleFunction } from '@typescript-eslint/utils/ts-eslint';
```

Comment:
Feels like the `RuleFunction` type is probably a little useful broadly. How about we export it normally?

```suggestion
  import type { RuleFunction } from '@typescript-eslint/utils';
```

Reply:
I don't love exporting it from multiple places just because it then gets confusing across a codebase.
"Did you mean to auto-import `RuleFunction` from `@typescript-eslint/utils` or `@typescript-eslint/utils/ts-eslint`?" Then you have half the codebase using one way and another half using the other way depending on who clicked what at the time.

Even worse you have 4 ways of consuming it:
```ts
import { TSESLint, RuleFunction } from '@typescript-eslint/utils';
import * as TSESLint2 from '@typescript-eslint/utils/ts-eslint';
import { RuleFunction as RuleFunction2 } from '@typescript-eslint/utils/ts-eslint';

// TSESLint.RuleFunction === TSESLint2.RuleFunction === RuleFunction === RuleFunction2
```

I think having the root namespaced export for a nice "single import declaration" case is good for those that want it (which is how we have already done things in the past), and having a single way to import names directly is good as well (which other codebases did a lot - via the `/dist/` path 😢).